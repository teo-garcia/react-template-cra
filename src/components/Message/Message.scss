.message {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;

  &__brand {
    font-size: 2.4rem;

    @media (min-width: 1024px) {
      font-size: 3rem;
    }
  }

  &__content {
    font-size: 2.4rem;
    margin: 32px 0 24px;

    @media (min-width: 1024px) {
      font-size: 3.6rem;
    }
  }

  &__subcontent {
    font-size: 1.5rem;

    @media (min-width: 1024px) {
      font-size: 1.8rem;
    }
  }
}
