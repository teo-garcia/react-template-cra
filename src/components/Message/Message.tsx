import './Message.scss';
import { MessageProps } from '../../tools/types';
import { Typography } from '../Typography/Typography';

function Message(props: MessageProps) {
  const { brandName, mainContent, secondaryContent } = props;

  return (
    <section className="message">
      <Typography className="message__brand" tagName="h1" weight="bold">
        {brandName}
      </Typography>
      <Typography className="message__content" color="neutral">
        {mainContent}
      </Typography>
      <Typography className="message__subcontent">
        {secondaryContent}
      </Typography>
    </section>
  );
}

export { Message };
