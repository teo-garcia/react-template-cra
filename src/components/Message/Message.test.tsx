import { render } from '@testing-library/react';
import { Message } from './Message';

describe('<Layout /> tests', function () {
  test('Should match with the component snapshot', function () {
    const { asFragment } = render(
      <Message
        brandName="Ping"
        mainContent="Lorem Ipsum"
        secondaryContent="Lorem ipsum"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
