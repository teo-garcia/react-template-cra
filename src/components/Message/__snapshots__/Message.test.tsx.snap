// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  <section
    class="message"
  >
    <h1
      class="typography typography--bold typography--text message__brand"
    >
      Ping
    </h1>
    <p
      class="typography typography--regular typography--neutral message__content"
    >
      Lorem Ipsum
    </p>
    <p
      class="typography typography--regular typography--text message__subcontent"
    >
      Lorem ipsum
    </p>
  </section>
</DocumentFragment>
`;
